server/db/session.py

```diff
@@ -1,28 +1,28 @@
+import logging
+
+from typing import AsyncGenerator
 from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine
-from sqlalchemy.ext.declarative import declarative_base
 
 from server.core.config import get_settings
 
 
 settings = get_settings()
 
-engine = create_async_engine(
-    settings.DATABASE_URL,
-)
+engine = create_async_engine(settings.DATABASE_URL, echo=True, future=True)
 
 AsyncSessionLocal = AsyncSession(
     autocommit=False, autoflush=False, bind=engine, future=True
 )
 
-Base = declarative_base()
-
-
-def get_db():
-    async def run():
-        db: AsyncSession = AsyncSessionLocal()
-        try:
-            yield db
-        finally:
-            await db.close()
 
-    return run
+async def get_db() -> AsyncGenerator[AsyncSession, None]:
+    db = AsyncSessionLocal
+    try:
+        logging.info("connecting to a database")
+        yield db
+        logging.info("Database connection - successful")
+    finally:
+        logging.info("Closing connection to database")
+        # TODO: investigate: INFO sqlalchemy.engine.Engine ROLLBACK
```

Comment:
Don't believe we'd need this for now as we're not continuously updating data in our application, only data we're using so far is from coastlines table (AFAIK). I know that we'd need to Rollback if we want to undo some recent changes we made to our DB prior to commitment.

Reply:
Are you responding to my TODO comment here? If yes, it's a note for myself. :) 

I observed this behavior when working on this PR and didn't understand why rollbacks were occurring at that particular point.